be/src/olap/rowset/segment_v2/segment_writer.cpp

```diff
@@ -261,6 +261,10 @@ Status SegmentWriter::_create_column_writer(uint32_t cid, const TabletColumn& co
         opts.data_page_size =
                 (page_size > 0) ? page_size : segment_v2::ROW_STORE_PAGE_SIZE_DEFAULT_VALUE;
     }
+    int64_t storage_page_size = _tablet_schema->storage_page_size();
```

Comment:
Change code for normal storage_page_size before row_store_page_size to avoid logic problem, since storage_page_size is default value and row_store_page_size is special value, special value should be processed after default value.